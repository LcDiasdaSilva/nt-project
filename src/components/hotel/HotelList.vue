<style lang="scss">
  .hotel-list{
    &__item{
      margin: 1rem 0;
    }
  }
</style>
<template>
  <div class="hotel-list">
    <div v-for="hotel in props.hotels" :key="hotel.id">
      <div class="hotel-list__item">
        <slot name="item" :hotel="hotel"></slot>
      </div>
    </div>
  </div>
</template>
<script setup lang="ts">
  import type { HotelInterface } from "../../models/interfaces";
  import { defineProps } from "Vue";

  const props = defineProps<{
    hotels: HotelInterface[];
  }>();
</script>
