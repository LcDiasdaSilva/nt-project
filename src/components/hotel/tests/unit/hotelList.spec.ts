import { shallowMount } from "@vue/test-utils";
import HotelList from "../../HotelList.vue";

const mockHotels = [
  {
    id: "1",
    name: "Hotel Bacana",
    description:
      "Este hotel à beira mar possui um spa e fica perto de Praia de Ipanema.",
    image: "/hotel1",
    address: {
      street: "Avenida Vieira Souto, 200",
      city: "Rio de Janeiro",
      state: "RJ",
      zip: "22420-002",
      country: "Brasil",
    },
    travelers: 1,
    bedrooms: 1,
    price: 900,
    reviews: [
      { note: 5 },
      { note: 2 },
      { note: 4 },
      { note: 3 },
      { note: 2 },
    ],
  },
  {
    id: "2",
    name: "Hotel Serenidade",
    image: "/hotel2",
    description:
      "Situado em uma área tranquila, este hotel oferece uma vista panorâmica para o campo.",
    address: {
      street: "Rua das Flores, 34",
      city: "Campos do Jordão",
      state: "SP",
      zip: "12460-000",
      country: "Brasil",
    },
    travelers: 2,
    bedrooms: 1,
    price: 600,
    reviews: [
      { note: 4 },
      { note: 5 },
      { note: 4 },
      { note: 4 },
      { note: 5 },
    ],
  },
];

describe("HotelList.vue", () => {
  it("matches snapshot when hotels list is not empty", () => {
    const wrapper = shallowMount(HotelList, {
      props: {
        hotels: mockHotels,
      },
      slots: {
        item: `<template #item="{ hotel }">
                 <div>{{ hotel.name }}</div>
               </template>`,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot when hotels list is empty", () => {
    const wrapper = shallowMount(HotelList, {
      props: {
        hotels: [],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
