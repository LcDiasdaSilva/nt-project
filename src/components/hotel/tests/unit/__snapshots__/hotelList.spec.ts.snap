// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList.vue matches snapshot when hotels list is empty 1`] = `
<div class="hotel-list">
  <div class="hotel-list__empty">Nenhum local encontrado</div>
</div>
`;

exports[`HotelList.vue matches snapshot when hotels list is not empty 1`] = `
<div class="hotel-list">
  <div>
    <div>
      <div class="hotel-list__item">
        <div>Hotel Bacana</div>
      </div>
    </div>
    <div>
      <div class="hotel-list__item">
        <div>Hotel Serenidade</div>
      </div>
    </div>
  </div>
</div>
`;
